conda/common/logic.py

```diff
@@ -433,7 +433,18 @@ def ExactlyOne(self, vals, polarity=None, name=None):
             what = self.ExactlyOne_NSQ
         else:
             what = self.ExactlyOne_BDD
-        return self._eval(what, (vals,), polarity, name)
+        return self._eval(what, (vals,), (), polarity, name)
+
+    def LinearBound(self, equation, lo, hi, preprocess=True, polarity=None, name=None):
+        if not isinstance(equation, dict):
+            # in case of duplicate literal -> coefficient mappings, always take the last one
+            equation = {named_lit: coeff for coeff, named_lit in equation}
```

Comment:
We did not really guard against duplicate keys in the `equation`/`objective` inputs. Now we at least detect duplicate strings or duplicate integers for the literals.
If we input `[(c1, 1), (c2, "v1")]` with `clauses.names["v1"] == 1`, we still do nonsense. Maybe we should fix that (easy enough).